### ISSUE TITLE
uprobes support

### ISSUE THREAD

**Issue by brendangregg** [iovisor/bcc#273]
Hopefully pretty similar to the existing kprobes support.

For now, it could have the separate attach interface; eg, in Python:

```Python
b.attach_uprobe(event="sleep", lib="/lib/x86_64-linux-gnu/libc.so.6" fn_name="trace_sleep")
```

Even better if the library could be provided as just "libc", "bash", etc. That would mean checking PATH first, then ldconfig. Here's some sample shell code to convert a shorthand name to the path:

```Shell
function set_path {
        name=$1

        path=$(which $name)
        if [[ "$path" == "" ]]; then
                path=$(ldconfig -v 2>/dev/null | awk -v lib=$name '
                    $1 ~ /:/ { sub(/:/, "", $1); path = $1 }
                    { sub(/\..*/, "", $1); }
                    $1 == lib { print path "/" $3 }')
                if [[ "$path" == "" ]]; then
                        die "ERROR: segment \"$name\" ambiguous." \
                            "Program or library? Try a full path."
                fi
        fi

        if [[ ! -x $path ]]; then
                die "ERROR: resolved \"$name\" to \"$path\", but file missing"
        fi
}
```

That's from https://github.com/brendangregg/perf-tools/blob/master/user/uprobe , which has examples of other bits required: writing the uprobe_events description.

**Comment by ceeaspb** [comment]
thanks for documenting this feature @brendangregg . it will certainly be useful for me

**Comment by brendangregg** [comment]
closed via #322.
